contracts/src/StakeTable.sol

```diff
@@ -99,18 +115,28 @@ contract StakeTable is AbstractStakeTable {
 
     /// @notice Get the next available epoch for exit
     function nextExitEpoch() external view override returns (uint64) {
-        if (numPendingExits == 0) {
-            return 0;
-        } else {
-            return 1;
-        }
+        //TODO
+        return currentEpoch() + 1;
     }
 
     /// @notice Get the number of pending exit requests in the waiting queue
     function numPendingExit() external view override returns (uint64) {
         return numPendingExits;
     }
 
+    /// @notice Defines the exit escrow period for a node.
+    /// TODO discuss Alex, Jeb. How much do we want to specify this function? Also marked as public
+    /// for easier testing.
+    /// @param node node which is assigned an exit escrow period.
+    /// @return Number of epochs post exit after which funds can be withdrawn.
+    function exitEscrowPeriod(Node memory node) public pure returns (uint64) {
+        if (node.balance > 100) {
+            return 10;
+        } else {
+            return 5;
+        }
+    }
```

Comment:
this should be the logic of `nextExitEpoch`, correct? I'm not sure why this exists.

Reply:
No, this is different. It is an extra period of time needed when withdrawing (after the exit is complete).
From the [specs](https://github.com/EspressoSystems/gitbook/blob/f3c8a00297dccb54b8f53cd255564f466e9a32dc/espresso-sequencer-architecture/internal-functionality/sequencer-contract/stake-table.md#exit):
> Once the exit epoch has been recorded, HotShot replicas will observe the change at the start of the next epoch, when they next synchronize the stake table, and honest replicas will consider the node exited once consensus enters exitEpoch. The node that exited will be able to withdraw their staked tokens after the escrow period has passed, using the withdraw function:

Reply:
Ah, thanks for the reference. let's add some code comments to explain this.

```
requestExit -- (queing) -->  Exited -- (escrow) --> Withdrawable
First phase is about waiting in queue due to rate-limiting on exit, the wait is dependent on the exit amount and currently exit traffic.
At the point of "Exited", Sequencer is officially off duty, and stops participating in consensus.
Second phase is about slashable security, the wait is dependent only on amount, during which period cryptographic evidence of misbehavior (e.g. double-voting) might still lead to the forfeit of stakes.
From the point of `Withdrawable` onwards, staker can freely withdraw.
```